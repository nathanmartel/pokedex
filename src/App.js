import React, { Component } from 'react';
import Header from './Header.js';
import PokeList from './PokeList.js';
import pokeTempData from './data-temp.js';
import request from 'superagent';
import './bootstrap-reboot.min.css';
import './style.css';
import './list.css';

export default class App extends Component {
  state = { 
    pokeData: []
  };

  async componentDidMount() {
    const data = await request.get('https://alchemy-pokedex.herokuapp.com/api/pokedex');
    this.setState({ pokeData: data.body.results });
  }

  render() {
    return (
      <div>
        <Header />
        {/* { JSON.stringify(this.state.pokeData) } */}
        <PokeList pokeData={this.state.pokeData} />
      </div>
    );
  }
}
