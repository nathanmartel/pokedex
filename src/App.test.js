import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('renders text "UniWhal" in the body', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/UniWhal/i);
  expect(linkElement).toBeInTheDocument();
});
