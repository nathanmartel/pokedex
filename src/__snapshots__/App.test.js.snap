// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div>
  <header>
    <h1
      className="hidden"
    >
      Pokemon
    </h1>
    <div
      class="img-container"
    >
      <img
        alt="Pokemon logo"
        src="pokemon-logo-png-2000.png"
      />
    </div>
  </header>
  <ul
    className="data-list"
  />
</div>
`;
